sdk/identity/Azure.Identity/tests/DefaultAzureCredentialTests.cs

```diff
@@ -84,6 +84,40 @@ public void ValidateCtorOptionsPassedToCredentials()
             Assert.AreEqual(expTenantId, actTenantId);
         }
 
+        [Test]
+        [NonParallelizable]
+        public void ValidateEnvironmentBasedOptionsPassedToCredentials([Values]bool clientIdSpecified, [Values]bool usernameSpecified, [Values]bool tenantIdSpecified)
+        {
+            var expClientId = clientIdSpecified ? Guid.NewGuid().ToString() : null;
+            var expUsername = usernameSpecified ? Guid.NewGuid().ToString() : null;
+            var expTenantId = tenantIdSpecified ? Guid.NewGuid().ToString() : null;
+
+            using (new TestEnvVar("AZURE_CLIENT_ID", expClientId))
```

Comment:
Tip: you can use the new `using var clientIdVar = new TestEnvVar(...);` syntax to have method scope here. Don't need to nest anything - even for the last block.